### DISCUSSION TITLE
Ensuring Consistent Critic Selection Mechanism from Multiple Response Options in Role-Playing Scenarios for Non-OpenAI Models

### DISCUSSION THREAD

**Discussion by coolbeevip** [camel-ai/camel#985] [General]
In the Role-Playing critic scenario, the user agent receives multiple response options through the n parameter of the OpenAI API, from which the critic selects one to guide the assistant agent. (If my understanding is incorrect, please feel free to point it out!)

However, this mechanism only works well with the OpenAI API; a brief investigation shows that other models do not support the n parameter. Should we ensure consistency in this behavior when using non-OpenAI models through engineering methods? (For example, by looping n times and concatenating the results.)

**Comment by Wendong-Fan** [comment]
Thanks, @coolbeevip! This is a fantastic idea—extending support for models that don't handle multiple responses at once. I raised an issue here: https://github.com/camel-ai/camel/issues/996. 
If you're interested in contributing, that would be also awesome!
